Makefile

```diff
@@ -0,0 +1,18 @@
+all: build
+
+dependencies:
+	go get -u github.com/golang/lint/golint
+	go get -u golang.org/x/tools/cmd/cover
+	go get -u golang.org/x/tools/cmd/vet
+
+build:
+	go fmt  ./...
+	golint ./...
+	go vet  ./...
+	go test -v -bench=.
+	go install ./...
+	go test -covermode=atomic -coverprofile=cover.out . \
+	&& go tool cover -html=cover.out -o=cover.html \
+	&& rm cover.out
+
+.PHONY: all dependencies build
```

Comment:
agreed, I don't understand the purpose of the subpackage either. Maybe it made more sense when there was an extra package too.

with `goclean`, you pretty much always want to use `./...`.
`gobench` works like most go tools: by default it operates on the current directory, but if you supply it a package name (or `./...`) it will run on that instead. So you can run `gobench ./...`.
`gocover` is a bit trickier. In most cases you only want to run it on one package at a time, but if you really want to run it on all packages, you can use `find` to list all the directories containing `.go` files:
```bash
find . -type f -name '*.go' | sed -r 's|/[^/]+$||' | sort | uniq
```